Sources/SwiftCliCore/OffsetUtility.swift

```diff
@@ -0,0 +1,87 @@
+import Foundation
+
+public struct RowInfo: Equatable {
+    var row: [String]
+    var rowNumber: Int
+    var bottomLine: String = ""
+    var stitchSymbols: String = ""
+    var width: Int = 0
+    var leftIncDec: Int = 0
+    var rightIncDec: Int = 0
+    // This is about the number of "empty stitches" that should be added on the left to account for future increases
+    var leftOffset: Int = 0
+    lazy var leftOffsetString: String = String(repeating: "  ", count: leftOffset)
+    lazy var offsetBottomLine: String = leftOffsetString + bottomLine
+    lazy var offsetStitchSymbols: String = leftOffsetString + stitchSymbols
+    lazy var totalRow: String = offsetStitchSymbols + offsetBottomLine
+}
+
+public class OffsetUtility {
```

Comment:
This name isn't too clear to me. `Utility` in general is a pretty vague term that can be replaced with something more precise. Secondly, this seems to be doing more than just determining offsets, but analyzing pattern metadata. See if you can find a name that accurately describes the responsibility of this class.

Reply:
Done! Renamed test file as well.